### PR TITLE
Add workflow template classes for contributions

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Add workflow template classes for contributions

Before
----------------------------------------
A big void

After
----------------------------------------
Tada

Technical Details
----------------------------------------
These aren't doing much of anyting yet - but they set up the structure - which will allow us to start to add in values to the trait (& remove some of those issets which mess with escaping)

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contribute/WorkflowMessage/ContributionTrait.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * @method array getContribution()
+ * @method array getContact()
+ * @method $this setContact(array $contact)
```
Comment (totten) [inline CRM/Contribute/WorkflowMessage/ContributionTrait.php:6]: Agree this PR looks safe / consistent / starting-place for other revisions. :+1:

I was about to hit "Merge" but then wondered about this fragment - `contact` is a property of `GenericWorkflowMessage` not `ContributionTrait`. 

There is another property in this trait, `contributionId` -- should it reference that instead?

```
 * @method ?int getContributionId()
 * @method $this setContributionId(?int $contributionId)
```

**Comment by eileenmcnaughton** [comment]
@totten you are right - I just updated to use contribution ID & updated one variable in one template to rely on that (plus test) to verify it

**Comment by eileenmcnaughton** [comment]
opps thanks @totten

**Comment by eileenmcnaughton** [comment]
Test fail is unrelated (date calc on the cusp of the new year - already started badly!)

**Pull request merged**